client/components/domains/register-domain-step/index.jsx

```diff
@@ -185,53 +152,51 @@ const RegisterDomainStep = React.createClass( {
 				this.showValidationErrorMessage( queryObject.query, domainError );
 			}
 		}
-	},
+	}
 
-	componentWillMount: function() {
+	componentWillMount() {
 		searchCount = 0; // reset the counter
-		lastSearchTimestamp = null; // reset timer
 
 		if ( this.props.initialState ) {
 			const state = { ...this.props.initialState, railcarSeed: this.getNewRailcarSeed() };
 
-			if ( state.lastSurveyVertical &&
-				( state.lastSurveyVertical !== this.props.surveyVertical ) ) {
+			if ( state.lastSurveyVertical && ( state.lastSurveyVertical !== this.props.surveyVertical ) ) {
 				state.loadingResults = true;
 				delete state.lastSurveyVertical;
 			}
 
 			this.setState( state );
 		}
-	},
 
-	componentDidMount: function() {
+		this._isMounted = false;
```

Comment:
Afaik, it will run the constructor again when it's created? Tho you might be right.